src/search/FindInFiles.js

```diff
@@ -417,15 +418,26 @@ define(function (require, exports, module) {
         }
         
         var scopeName = searchModel.scope ? searchModel.scope.fullPath : ProjectManager.getProjectRoot().fullPath,
-            perfTimer = PerfUtils.markStart("FindIn: " + scopeName + " - " + queryInfo.query);
+            perfTimer = PerfUtils.markStart("FindIn: " + scopeName + " - " + queryInfo.query),
+            searchDeferred = new $.Deferred();
         
-        return candidateFilesPromise
+        candidateFilesPromise
             .then(function (fileListResult) {
                 // Filter out files/folders that match user's current exclusion filter
                 fileListResult = FileFilters.filterFileList(filter, fileListResult);
+                fileListResult = FindUtils.getSortedFiles(fileListResult, function (file) { return file.fullPath; }, FindUtils.getOpenFilePath());
```

Comment:
@TomMalbran `SearchModel` stores the results as an object with path as the key. `SearchResultsView` needs to re-sort list anyway because iterating over properties of an object does not guarantee any specific order. Also, file searching is done in parallel within batches, so there's no guarantee that search results are returned in sort order. So, sorting needs to be done in the `SearchResultsView`, and it looks like it's already done in `SearchResultsView._render`.

So, order of results added to panel as search progresses may not be linear. Sorting files here may *reduce* (not eliminate) that, but unless we can figure out what the slowdown is we shouldn't do it. Just moving current doc to top of list might be good enough.